crates/ruff/src/rules/pyupgrade/rules/native_literals.rs

```diff
@@ -165,13 +174,20 @@ pub(crate) fn native_literals(
 
     match args.get(0) {
         None => {
-            let mut diagnostic = Diagnostic::new(NativeLiterals { literal_type }, expr.range());
+            let mut diagnostic = Diagnostic::new(NativeLiterals { literal_type }, call.range());
+
+            // Do not suggest fix for attribute access on an int like `int().attribute`
+            // Ex) `int().denominator` is valid but `0.denominator` is not
```

Comment:
I assume floats and complex are fine?

Reply:
I was a little scared to test them and get sucked into more issues, but I'll look into it.